build/Dependencies.props

```diff
@@ -15,7 +15,7 @@
   <PropertyGroup>
     <GoogleProtobufPackageVersion>3.10.1</GoogleProtobufPackageVersion>
     <LightGBMPackageVersion>2.2.3</LightGBMPackageVersion>
-    <MicrosoftExtensionsPackageVersion>2.1.0</MicrosoftExtensionsPackageVersion>
+    <MicrosoftExtensionsPackageVersion>3.0.1</MicrosoftExtensionsPackageVersion>
```

Comment:
How will this affect users who are targeting .NET Core 2.1 in their web apps? Is it valid for us to pull in a higher `Microsoft.Extensions.*` assemblies? I believe it will mean that these assemblies will now be published along with the application, and no longer used from the shared framework, which means larger apps. It could also mean slower start up times because the assemblies won't be crossgen'd.

/cc @glennc

Reply:
@glennc @davidfowl @Pilchie - is the `OnChange` bug fix for dotnet/extensions#558 something that should be ported to `2.1`? I'm concerned here that anyone using `2.1` + ML.NET would be forced to update `Microsoft.Extensions.*` to `3.0.1`. If this is a serious enough bug to cause ML.NET to force everyone to `3.0`, maybe it should be back-ported to `2.1.` instead.

Reply:
In general, we don't really recommend using 3.x NuGet packages in a 2.1 app.  It might work for this one, but there are a significant number of breaking changes in ASP.NET all-up that makes it problematic.  We can consider back-porting a fix to 2.1, but the usual bar is that there is a large customer blocked by the issue.

Reply:
What is the impact of this bug on ML.NET Customer?

Reply:
@sharwell I thought you said you noticed this issue in the LdaTransform. Can you please help identify the overall impact?

Reply:
> What is the impact of this bug on ML.NET Customer?

@sharwell - can you answer this?

Reply:
> What is the impact of this bug on ML.NET Customer?

@sharwell - can you answer this?

Reply:
The problem was observed here:

https://github.com/dotnet/machinelearning/blob/f3e0f6b3a52f7e3998978239a0f52f35fd6197d8/test/Microsoft.Extensions.ML.Tests/UriLoaderTests.cs#L44-L48

Disposing of the registration doesn't do anything. It's been a while since I reviewed this to figure out all the details. What about conditioning this dependency on the build target, so we have a dependency to 2.1 for netcoreapp2.1, but a dependency on 3.0.1 when targeting netcoreapp3.0+?

Reply:
If this is just a test concern, I think we should update the test's .csproj to reference `Microsoft.Extensions.*` version `3.0.*` as necessary.

Users who need this bug fix can reference `3.0` themselves, if necessary. But at least we won't force any users on `2.1` to be in a non-recommended situation.

Reply:
@sharwell Are there other cases where this bug surfaces? If not, is Eric's suggestion an acceptable fix?

Reply:
@eerhardt Does the update PR address your concerns? If so, can you please approve?